packages/orchestration/scripts/fetch-chain-info.ts

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env tsx
+import fsp from 'node:fs/promises';
+import assert from 'node:assert';
+import { ChainRegistryClient } from '@chain-registry/client';
+import { State as IBCConnectionState } from '@agoric/cosmic-proto/ibc/core/connection/v1/connection.js';
+import type { IBCChannelID, IBCConnectionID } from '@agoric/vats';
+import {
+  Order,
+  State as IBCChannelState,
+} from '@agoric/cosmic-proto/ibc/core/channel/v1/channel.js';
+import type { IBCInfo } from '@chain-registry/types';
+import type { CosmosChainInfo, IBCConnectionInfo } from '../src/cosmos-api.js';
+
+// XXX script assumes it's run from the package path
+// XXX .json would be more apt but Endo bundler can't import that
+const outputFile = 'src/fetched-chain-info.js';
+
+/**
+ * Names for which to fetch info
+ */
+const chainNames = [
+  'celestia',
+  'cosmoshub',
+  'dydx',
+  'juno',
+  'neutron',
+  'noble',
+  'omniflixhub',
+  'osmosis',
+  'secretnetwork',
+  'stargaze',
+  'stride',
+];
+
+const client = new ChainRegistryClient({
+  chainNames,
+});
+
+// chain info, assets and ibc data will be downloaded dynamically by invoking fetchUrls method
+await client.fetchUrls();
+
+function toConnectionEntry(ibcInfo: IBCInfo, name: string) {
+  // IbcInfo encodes the undirected edge as a tuple of (chain_1, chain_2) in alphabetical order
+  const fromChain1 = ibcInfo.chain_1.chain_name === name;
+  const [from, to] = fromChain1
+    ? [ibcInfo.chain_1, ibcInfo.chain_2]
+    : [ibcInfo.chain_2, ibcInfo.chain_1];
+  assert.equal(from.chain_name, name);
+  const transferChannels = ibcInfo.channels.filter(
+    c =>
+      c.chain_1.port_id === 'transfer' &&
+      // @ts-expect-error tags does not specify keys
+      c.tags?.preferred,
+  );
+  if (transferChannels.length > 1) {
+    console.warn(
+      'multiple preferred transfer channels [',
+      from.chain_name,
+      to.chain_name,
+      ']:',
+      transferChannels,
+      '(choosing first)',
+    );
+  }
+  const [channel] = transferChannels;
+  const [channelFrom, channelTo] = fromChain1
+    ? [channel.chain_2, channel.chain_1]
+    : [channel.chain_1, channel.chain_2];
+  const record = {
+    id: from.connection_id as IBCConnectionID,
+    client_id: from.client_id,
+    counterparty: {
+      client_id: to.client_id,
+      connection_id: to.connection_id as IBCConnectionID,
+      prefix: {
+        key_prefix: 'FIXME',
+      },
+    },
+    state: IBCConnectionState.STATE_OPEN, // XXX presumably
+    transferChannel: {
+      channelId: channelFrom.channel_id as IBCChannelID,
+      portId: channelFrom.port_id,
+      counterPartyChannelId: channelTo.channel_id as IBCChannelID,
+      counterPartyPortId: channelTo.port_id,
+      // FIXME mapping, our guard expects a numerical enum
+      ordering: Order.ORDER_NONE_UNSPECIFIED,
+      state: IBCChannelState.STATE_OPEN, // XXX presumably
+      version: channel.version,
+    },
+    versions: [], // XXX where to read?
```

Comment:
@0xpatrickdev prototyped code to get this stuff from the protocol

https://github.com/Agoric/agoric-sdk/issues/8879#issuecomment-2123142794
https://github.com/Agoric/agoric-sdk/issues/8879#issuecomment-2123582330

If we're not getting the info from the protocol, then it's not clear that we should use this structure.

Reply:
I think we should remove this field until there's a story that requires it.


Making example data more visible for the links comments:
```
 {
    "identifier": "1",                   // seems like it'd only come into play when there's a v2 of `ibc`. 
    "features": [
      "ORDER_ORDERED",        // doesn't seem like something needed at run-time
      "ORDER_UNORDERED"
    ]
  }
```